src/test/lambda/local/detectLocalTemplates.test.ts

```diff
@@ -0,0 +1,295 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { detectLocalTemplates } from '../../../lambda/local/detectLocalTemplates'
+import * as filesystem from '../../../shared/filesystem'
+
+function normalizePath(...paths: string[]): string {
+    return vscode.Uri.file(path.join(...paths)).fsPath
+}
+
+describe('detectLocalTemplates', async () => {
+    it('Detects no templates when there are no workspace folders', async () => {
+        for await (const template of detectLocalTemplates()) {
+            assert.fail(`Expected no templates, but found '${template.fsPath}'`)
+        }
+    })
+
+    it('Detects templates at the root of each workspace folder', async () => {
+        const workspaceFolderPath = normalizePath('my', 'workspace', 'folder')
+
+        const result = detectLocalTemplates(
+            {
+                async accessAsync(_path: string | Buffer): Promise<void> {
+                    if (_path !== normalizePath(workspaceFolderPath, 'template.yaml')) {
+                        throw new Error(`No file found at path: '${_path}'`)
+                    }
+                },
+
+                async readdirAsync(
+                    _path: string | Buffer,
+                    options?: {
+                        encoding: BufferEncoding | null
+                        withFileTypes?: false
+                    } | BufferEncoding | undefined | null
+                ): Promise<string[]> {
+                    return [ 'template.yaml' ]
+                },
+
+                async statAsync(
+                    _path: string | Buffer
+                ): Promise<filesystem.Stats> {
+                    return {
+                        isDirectory() {
+                            return true
+                        }
+                    } as any as filesystem.Stats
+                }
+            },
+            {
+                uri: vscode.Uri.file(workspaceFolderPath)
+            } as any as vscode.WorkspaceFolder
```

Comment:
There are a few places that I missed using `Pick<vscode.WorkspaceFolder, 'uri'>` instead of `vscode.WorkspaceFolder`. I'll fix them for the next iteration.